sinon/index.d.ts

```diff
@@ -427,7 +427,7 @@ declare namespace Sinon {
 
     // Utility overridables
     interface SinonStatic {
-        createStubInstance(constructor: any): SinonStub;
+        createStubInstance(constructor: any): any;
```

Comment:
@andy-ms I realized that proposed typing will still be inaccurate, as `createStubInstance` makes only function properties to be of `SinonStub` type:
```js
a = function() {}
a.prototype.b = 3;
a.prototype.c = function() {};
stub = sinon.createStubInstance(a);
console.log(typeof stub.c.getCall); // 'function', method of SinonStub
console.log(typeof stub.b.getCall); // 'undefined'
console.log(typeof stub.b); // 'number' - b is still number, not SinonStub
```
As far as I see, this can't be achieved now and will only be achievable with mapped conditional types - [feature request](https://github.com/Microsoft/TypeScript/issues/12424),  [original suggestion](https://github.com/Microsoft/TypeScript/pull/12114#issuecomment-259776847)

So I'd say, let's leave it `any` for now and get back to it when it can be properly typed with mapped types.